test-strategy.md

```diff
@@ -0,0 +1,108 @@
+# Test Strategy
+
+Very brief, mildly aspirational test strategy document. This isn't where we are but it is where we want to get to.
+
+This does not detail how to setup an environment or how to run the tests locally nor does it contain any best practices that we try to follow when writing tests, that information exists in an as yet unwritten contributing guide (ye, i know).
```

Comment:
We do have a contributing document that does contains how to setup an environment or how to run the tests locally: https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md

Can we link from this doc?

Reply:
Done both here and in the more detailed section below.